packages/wallet/src/Wallets/BaseWallet.ts

```diff
@@ -821,6 +843,41 @@ export class BaseWallet implements ObservableWallet {
     return firstValueFrom(this.addresses$);
   }
 
+  async getNextUnusedAddress(): Promise<WalletAddress | null> {
```

Comment:
Should we derive 1 per stake key? :thinking: 
```suggestion
  async getNextUnusedAddresses(): Promise<WalletAddress[]> {
```

Reply:
Good point @mkazlauskas. This is probably the most appropriate thing to do but, with our current data API and fetching strategy to catch rollbacks, we need to be careful not to place too much additional load on the DB. We can always extend this with an improved API but being cautious for now seems appropriate.

Reply:
Per our multi-delegation convention, secondary stake keys should only ever be paired with the first payment key, only the first stake key (index 0) is allowed to have more than one payment credential.